web/src/types/option.ts

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable unicorn/prevent-abbreviations */
+import { ToString } from './utils';
+import { Err, Ok, Result } from './result';
+
+export const Some = <S extends ToString>(value: S) => {
+  return new Option<S>(false, value);
+};
+
+export const None = () => {
+  return new Option<never>(true);
+};
+
+export class Option<S extends ToString> {
+  private is_none: boolean;
```

Comment:
I don't know, I think I'd prefer to use subtypes for this rather than boolean flags, but I'm guessing there's a reason it's done this way?

Reply:
Option now uses subtypes. Changed in bad24a5